sled-agent/src/rack_setup/service.rs

```diff
@@ -689,14 +693,16 @@ impl ServiceInner {
                             dns_address,
                         } => {
                             services.push(NexusTypes::ServicePutRequest {
-                                service_id: zone.id,
+                                service_id: Uuid::new_v4(),
```

Comment:
I feel like this Uuid generation shouldn't happen here (`handoff_to_nexus`) but somewhere earlier as part of creating the allocation plan. Otherwise, next boot we'll see the previously completed plan and still end up here but end up passing a slightly different `ServicePutRequest` to Nexus.

Reply:
Good catch!  I've fixed this in 901c04a.  Let me know what you think.  I basically replaced the list of `ServiceKind` with a list of a new struct containing both the `ServiceKind` and a unique id.  On the one hand, this feels overly complex.  But I don't see a way around it without more substantial refactoring.

One surprising consequence of this is that there are places _inside_ the Sled Agent where we pass around a `ServicePutRequest` that was manufactured inside the Sled Agent (e.g., `activate_switch()`).  These structs are changed in this PR and now get these extra ids.  That's even though as far as I can tell, these never wind up registered in Nexus.  I think this is an artifact of using the same struct to represent "a request to provision a zone with one or more services in it" for two different purposes: one is to report it to Nexus as a `service`, and the other is to actually cause that zone to be provisioned.  The former requires these extra ids.  The latter doesn't, but winds up getting them.